disperser/batcher/batcher.go

```diff
@@ -316,8 +316,11 @@ func (b *Batcher) HandleSingleBatch(ctx context.Context) error {
 		}
 
 		if status == disperser.Confirmed {
-			_, updateConfirmationInfoErr = b.Queue.MarkBlobConfirmed(ctx, metadata, confirmationInfo)
-			b.Metrics.UpdateCompletedBlob(int(metadata.RequestMetadata.BlobSize), disperser.Confirmed)
+			if _, updateConfirmationInfoErr = b.Queue.MarkBlobConfirmed(ctx, metadata, confirmationInfo); updateConfirmationInfoErr == nil {
+				b.Metrics.UpdateCompletedBlob(int(metadata.RequestMetadata.BlobSize), disperser.Confirmed)
+				// remove encoded blob from storage now that the blob is confirmed
+				b.EncodingStreamer.RemoveEncodedBlob(metadata)
+			}
 		} else if status == disperser.InsufficientSignatures {
 			_, updateConfirmationInfoErr = b.Queue.MarkBlobInsufficientSignatures(ctx, metadata, confirmationInfo)
```

Comment:
Is this not a terminal state? Should we remove the blobs for this one, also?

Reply:
Yes! I'll update